src/new_hope/week1.clj

```diff
@@ -48,5 +48,116 @@
   ;; More elaborate benchmark taking 10 seconds
   (let [coll (vec (range 10000))]
     (crit/quick-bench (second (reverse coll)))
-    (crit/quick-bench (last (butlast coll))))
-  )
+    (crit/quick-bench (last (butlast coll)))))
+
+
+;; http://www.4clojure.com/problem/24
+(def sum-it-all-up
+  (fn [coll]
+    (apply + coll)))
+
+(facts "about sum-it-all-up"
+       (sum-it-all-up [1 2 3]) => 6
+       (sum-it-all-up (list 0 -2 5 5)) => 8
+       (sum-it-all-up #{4 2 1}) => 7
+       (sum-it-all-up '(0 0 -1)) => -1
+       (sum-it-all-up '(1 10 3)) => 14)
+
+;;http://www.4clojure.com/problem/25
+(def filter-odd
+  (fn [coll]
+    (filter odd? coll)))
+
+(facts "about filter-odd"
+       (filter-odd #{1 2 3 4 5}) => '(1 3 5)
+       (filter-odd [4 2 1 6]) => '(1)
+       (filter-odd [2 2 4 6]) => '()
+       (filter-odd [1 1 1 3]) => '(1 1 1 3))
+
+;;http://www.4clojure.com/problem/27
+(def palindrome?
+  (fn [coll]
+    (= (apply str (reverse coll)) (apply str coll))))
```

Comment:
No need to apply str here. Comparing collections directly is more efficient. For dealing with "racecar" case convert `coll` to a collection.